cpp/src/parquet/arrow/schema.cc

```diff
@@ -428,13 +447,19 @@ Status FieldToNode(const std::string& name, const std::shared_ptr<Field>& field,
     }
     case ArrowTypeId::EXTENSION: {
       auto ext_type = std::static_pointer_cast<::arrow::ExtensionType>(field->type());
-      // Built-in JSON extension is handled differently.
+      // Built-in JSON extension and GeoArrow are handled differently.
```

Comment:
I was thinking if we can add an extension type registry which allows users to register an extension name and the function to create the extension type and/or parquet type. This is similar to what I've added to avro recently: https://github.com/apache/avro/pull/3326/files#diff-00521bd02440832a207504b570a0fc02ccbe0928c6710e419503d48284aeee90R95-R114